gleam_stdlib/test/list_test.gleam

```diff
@@ -248,3 +250,14 @@ pub fn sort_test() {
   list:sort([{1,2}, {4,5}, {3,2}])
   |> expect:equal(_, [{1,2}, {3,2}, {4,5}])
 }
+
+pub fn index_map_test() {
+  list:index_map([3,4,5], fn(i, x) { {i, x} })
+  |> expect:equal(_, [{0,3},{1,4},{2,5}])
+
+  let f = fn(i, x) {
+    iodata:new(x) |> iodata:append(_, str:from_int(i)) |> iodata:to_string(_)
```

Comment:
Is there a better way to do this? I'm about to do something like `str:join` but maybe I'm missing something obvious about the Gleam/Erlang world